csrc/python_frontend/python_bindings.cpp

```diff
@@ -555,49 +632,23 @@ void initNvFuserPythonBindings(PyObject* module) {
               {self.recordingState(out())}, std::monostate{}, DataType::Int));
           args.emplace_back(out);
         }
-        return define_vector_fn(self, args);
+        return define_vector_base_fn(self, args);
       },
       py::arg("size"),
       py::return_value_policy::reference);
   // This is the constant version of define_vector when given a vector
   // of constant values.
   fusion_def.def(
       "define_vector",
-      [](FusionDefinition& self, py::list& values) -> Vector {
-        std::vector<Scalar> args;
-        size_t idx = 0;
-        for (const auto& item : values) {
-          TORCH_CHECK(
-              idx < 8,
-              "The specified vector size exceeds the max tensor size for nvfuser.");
-          if (py::isinstance<py::int_>(item)) {
-            auto int_value = py::cast<int64_t>(item);
-            TORCH_CHECK(
-                int_value >= -1,
-                "The value ",
-                int_value,
-                " at index ",
-                idx,
-                " was neither symbolic(-1), zero_element(0), broadcast(1), or static(>1).");
-            Scalar out = self.defineScalar();
-            self.defineRecord(new ScalarRecord(
-                {self.recordingState(out())},
-                py::cast<int64_t>(item),
-                DataType::Int));
-            args.emplace_back(out);
-          } else if (py::isinstance<Scalar>(item)) {
-            args.emplace_back(py::cast<Scalar>(item));
-          } else {
-            TORCH_CHECK(
-                false,
-                "Unsupported iterable object type for define_vector! Index:",
-                idx);
-          }
-          ++idx;
-        }
-        return define_vector_fn(self, args);
-      },
+      define_vector_fn<py::list>,
       py::arg("values"),
+      py::arg("dtype") = DataType::Int,
```

Comment:
I believe this means we can now create non-int `Vector`s right? Nice.

Reply:
Not exactly.  The possibility exists but the reality is that it will assert as `Vector` only represents sizes, currently.

Reply:
Do we need error testing for this case? e.g., passing a `Vector` of floats to `broadcast_in_dims`

Reply:
Really you are asking if `define_vector` needs the test as the `broadcast_in_dim` python function is a composition of Fusion Records.  The answer is probably "yes" for completeness.